clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -83,6 +85,15 @@ public void configure(Map<String, ?> configs) throws KafkaException {
         if (endpointIdentification != null)
             this.endpointIdentification = endpointIdentification;
 
+        String secureRandomImplementation = (String) configs.get(SslConfigs.SSL_SECURE_RANDOM_IMPLEMENTATION_CONFIG);
+        if (secureRandomImplementation != null) {
+            try {
+                this.secureRandomImplementation = SecureRandom.getInstance(secureRandomImplementation);
+            } catch (GeneralSecurityException e) {
```

Comment:
Another option if we want to default to SHA1PRNG is to catch `NoSuchAlgorithmException` and fall back to `null` (with a warning).